Objects/bytes_methods.c

```diff
@@ -556,11 +556,56 @@ find_internal(const char *str, Py_ssize_t len,
     return res;
 }
 
+#define FIND_CHUNK_SIZE 10000
+#define RFIND_CHUNK_SIZE FIND_CHUNK_SIZE
+
 PyObject *
-_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *sub,
+_Py_bytes_find(const char *str, Py_ssize_t len, PyObject *subobj,
                Py_ssize_t start, Py_ssize_t end)
 {
-    Py_ssize_t result = find_internal(str, len, "find", sub, start, end, +1);
+    Py_ssize_t result;
+    if (PyTuple_Check(subobj)) {
+        result = -1;
+        ADJUST_INDICES(start, end, len);
+        // Work in chunks
+        for (; result == -1 && start <= end; start += FIND_CHUNK_SIZE) {
+            Py_ssize_t cur_end = start + FIND_CHUNK_SIZE;
+            for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(subobj); i++) {
+                PyObject *subseq = PyTuple_GET_ITEM(subobj, i);
+                Py_ssize_t sub_len;
+                Py_buffer subbuf;
+                if (!PyObject_CheckBuffer(subseq)) {
+                    sub_len = 1;
+                }
+                else if (PyObject_GetBuffer(subseq, &subbuf,
+                                            PyBUF_SIMPLE) != 0)
+                {
+                    return NULL;
+                }
+                else {
+                    sub_len = subbuf.len;
+                }
+                Py_ssize_t sub_end = cur_end + sub_len;
```

Comment:
```suggestion
                Py_ssize_t sub_end = cur_end + sub_len - 1;
```

Reply:
Let's change `cur_end` instead: `Py_ssize_t cur_start = cur_end - RFIND_CHUNK_SIZE + 1`.
That makes it easier to follow (and might be faster).

Reply:
To me it is more intuitive other way, but this works too.

Reply:
The start of the substring lays in `[start:start+FIND_CHUNK_SIZE-1]`, doing it this way makes it clear that this range is for the start.
The end of the substring lays in `[start+len(substr):start+FIND_CHUNK_SIZE-1+len(substr)]`.
I think that's quite intuitive.